### PR TITLE
Use _LP64 to check for 64-bit platforms

### PR DESCRIPTION
**Pull request by hack3ric**
Allows more 64-bit platforms other than x86_64 to build on Linux and more.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Doesn't this break building on x86-64 that don't use LP64.

**Comment by ZehMatt** [comment]
This change also seems rather random, is there a problem with the current approach?

**Comment by hack3ric** [comment]
Almost all 64-bit Linux systems and all 64-bit macOS ones use LP64. Windows uses LLP64, however, but that is covered by `_M_X64` and `_M_ARM64`. One possibly safer way is to enumerate all 64-bit platforms:

```
#if defined(_M_X64) || defined(_M_ARM64) || defined(__x86_64__) || defined(__aarch64___) || defined(__ppc64__) || (defined(__riscv) && __riscv_xlen == 64)
```

(AFAIK those non-LP64 Linux platforms include, for example x32, which uses ILP32 data model and 32-bit pointer size)

**Comment by ZehMatt** [comment]
> Almost all 64-bit Linux systems and all 64-bit macOS ones use LP64. Windows uses LLP64, however, but that is covered by `_M_X64` and `_M_ARM64`. One possibly safer way is to enumerate all 64-bit platforms:
> 
> ```
> #if defined(_M_X64) || defined(_M_ARM64) || defined(__x86_64__) || defined(__aarch64___) || defined(__ppc64__) || (defined(__riscv) && __riscv_xlen == 64)
> ```
> 
> (AFAIK those non-LP64 Linux platforms are, for example x32, using ILP32 data model and 32-bit pointer size)

That doesn't really answer my question however. Also do we need to specify absolutely all platforms if we could just check for LP64 or LLP64?

**Comment by hack3ric** [comment]
> That doesn't really answer my question however. Also do we need to specify absolutely all platforms if we could just check for LP64 or LLP64?

Sorry about the confusion. I would like to see the project to build on more platforms, while current method only allows x86_64 to build on Linux.

**Comment by duncanspumpkin** [comment]
What about we just change this to a check for `PLATFORM_64BIT` from common.h (which is just a `#if defined(__LP64__) || defined(_WIN64)`)

**Comment by ZehMatt** [comment]
> > That doesn't really answer my question however. Also do we need to specify absolutely all platforms if we could just check for LP64 or LLP64?
> 
> Sorry about the confusion. I would like to see the project to build on more platforms, while current method only allows x86_64 to build on Linux.

The platform has to also ideally exist on GitHub so we can run automated builds and tests on the CI.

**Comment by tupaschoal** [comment]
I'd rephrase the question as: "is there a known platform where you failed to build?"

Because for sure we want to support as much platforms as we can, but we'll never be able to support them all and specially not test them all.

**Comment by tupaschoal** [comment]
I don't think it is, the last comment was a question from me

**Comment by hack3ric** [comment]
Sorry about missing on those questions. I wanted to build OpenRCT on riscv64, and with this relaxed macro condition I was able to make it. I wonder if we can just allow those unsupported platforms to build, while not breaking the supported ones (which I am pretty sure this change won't)?

**Comment by ZehMatt** [comment]
It's probably better to move the platform detection code into Platform.h and move that stuff out of common.h and then use in BitSetTests the appropriate updated ones.
